### DISCUSSION TITLE
Migrating from Windows deployment to HA Addon

### DISCUSSION THREAD

**Discussion by peca89** [Koenkk/zigbee2mqtt#23867] [General]
TL;DR - How can I migrate from standalone NPM installation on Windows to new HA Addon deployment with keeping all the data and moving the same physical USB stick?

I’ve been using z2m for years running natively on a Windows machine using npm for Windows (because reasons beyond this post). That machine has suddenly died. Now I want to migrate my HA+z2m instance to the Virtualization Station on QNAP NAS. Installing HA was easy because I had previous HA instance create weekly backups and auto-upload them to the NAS. But installing z2m addon is a problem now because previous z2m instance on Windows was very old. Z2m seems to support [restoring data from a standalone installation](https://github.com/zigbee2mqtt/hassio-zigbee2mqtt?tab=readme-ov-file#restoring-data-from-a-standalone-installation) but this requires versions to match. It might work accross versions, but I’m not risking having to re-pair 30+ devices, some of which inaccessible physically. I also might be able to spin up another Windows machine from backup and update npm version there, but previous attempts to run npm update commands on Windows have usually broken something. 

Options that I see:

1. Install matching addon version (1.32.1), migrate data, update addon - I don't know how to install a specific addon version instead of the latest
2. Redeploy Windows host, update to the latest version there, then migrate to addon - I don't know anything about how NPM functions and updating z2m previously on Windows using `npm ci` and similar commands has always broken something
3. Ignore version mismatch and try just to migrate data - I would like not to take any risk in having to re-pair zigbee devices.

**Comment by Koenkk** [comment]
Copying over the `data` directory + running latest z2m in HA addon is enough.

**Reply by peca89** [reply]
Worked fine. Thanks you!

Just maybe a suggestion to modify instructions at [restoring data from a standalone installation](https://github.com/zigbee2mqtt/hassio-zigbee2mqtt?tab=readme-ov-file#restoring-data-from-a-standalone-installation). It seems that SSHing directly to the host is not needed anymore. Z2m's data folder is now visible from File Editor addon at `/homeassistant/zigbee2mqtt` and also through Samba addon at `\\x.x.x.x\config\zigbee2mqtt\`

**Reply by Koenkk** [reply]
Nice, could you make a PR by clicking the edit button?
